my-contacts/src/Contacts.js

```diff
@@ -0,0 +1,101 @@
+import React, {useEffect, useState} from 'react';
+import Contact from "./Contact.js";
+import Form from "./Form";
+import PropTypes from "prop-types";
+import {useStyles} from "react-treat";
+import * as styleRefs from "./Contacts.treat";
+
+
+const Contacts = (props) => {
+
+    const styles = useStyles(styleRefs);
+
+    const [adding, setAdding] = useState(false);
+    const [filteredContacts, setFilteredContacts] = useState(props.contacts);
+
+    useEffect(() => {
+        setFilteredContacts(props.contacts);
+    }, [props.contacts]);
+
+    const onButtonPressed = () => {setAdding(!adding)};
+
+    const contactCreated = contact => {
+        setAdding(false);
+        props.addContact(contact)
+    };
+
+    const contactMatchesSearch = (contact, search) => {
+        search = search.toLowerCase();
+
+        return contact.lastName.toLowerCase().startsWith(search) || contact.firstName.toLowerCase().startsWith(search)
+    };
+
+    const search = e => {
+        let q = e.target.value;
+        let filteredContacts = props.contacts.filter(contact => contactMatchesSearch(contact, q));
+        setFilteredContacts(filteredContacts)
+    };
+
+    const buttonText = adding ? "Cancel" : "Add";
+
+    const header = (
+        <div>
+            <div className={styles.titleContainer}>
+                <h1 className={styles.title}>My contacts</h1>
+                <div className={styles.buttonContainer}>
+                    <button onClick={onButtonPressed}
+                            className="add"
+                            >{buttonText}</button>
+                </div>
+            </div>
+
+            <hr className={styles.line}/>
+        </div>
+    );
+
+    const content = filteredContacts.map(contact => {
```

Comment:
Could probably move this into the else part  so we dont have do all the logic to build it then not use it.